sdk/storage/azure-storage-common/src/xml_wrapper.cpp

```diff
@@ -35,17 +35,18 @@ namespace Azure { namespace Storage { namespace Details {
     }
   }
 
-  XmlReader::~XmlReader() { xmlFreeTextReader(m_reader); }
+  XmlReader::~XmlReader() { xmlFreeTextReader(static_cast<xmlTextReaderPtr>(m_reader)); }
```

Comment:
Add a comment here (and elsewhere `static_cast` is used) to explain/justify why it is safe and OK to do this as a reminder to future maintainers.